src/vm/stackwalk.cpp

```diff
@@ -1778,7 +1778,7 @@ StackWalkAction StackFrameIterator::Filter(void)
                                 pCurrTracker = pCurrTracker->GetPreviousExceptionTracker();
                             }
 
-                            if (pCurrTracker != NULL)
+                            if ((pCurrTracker != NULL) && !pCurrTracker->IsEHClauseFilter())
```

Comment:
I think I was debugging this test

JIT/Methodical/eh/basics/throwinfilter_d/throwinfilter_d.exe

The issue happened in the nested call to IL_Throw.